Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,318 @@
+        bool fileExists = AZ::IO::FileIOBase::GetInstance()->Exists(prefabPath.c_str());
+
+        if (CheckCyclicalDependency(prefabPathRealative))
+        {
+            m_prefabMakerPage->setSuccess(false);
+            return;
+        }
+        if (fileExists)
+        {
+            QMessageBox msgBox;
+            msgBox.setText(tr("Prefab with this name already exists"));
+            msgBox.setInformativeText(tr("Do you want to overwrite existing prefab?"));
+            msgBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Cancel);
+            msgBox.setDefaultButton(QMessageBox::Cancel);
+            int ret = msgBox.exec();
+            if (ret == QMessageBox::Cancel)
+            {
+                m_prefabMakerPage->setSuccess(false);
+                return;
+            }
+        }
+        m_prefabMaker = AZStd::make_unique<URDFPrefabMaker>(m_urdfPath, m_parsedUrdf, prefabPath.String(), m_urdfAssetsMapping);
+
+        auto callback = [&]()
+        {
+            emit SignalFinalizeURDFCreation();
+        };
+        m_prefabMaker->LoadURDF(callback);
+    }
+    void RobotImporterWidget::FinalizeURDFCreation()
+    {
+        auto prefabOutcome = m_prefabMaker->CreatePrefabFromURDF();
+        if (prefabOutcome.IsSuccess())
+        {
+            AZStd::string status = m_prefabMaker->getStatus();
+            m_prefabMakerPage->reportProgress(status);
+            m_prefabMakerPage->setSuccess(true);
+        }
+        else
+        {
+            AZStd::string status = "Failed to create prefab\n";
+            status += prefabOutcome.GetError() + "\n";
+            status += m_prefabMaker->getStatus();
+            m_prefabMakerPage->reportProgress(status);
+            m_prefabMakerPage->setSuccess(false);
+        }
+    }
+    void RobotImporterWidget::onCreateButtonPressed()
+    {
+        CreatePrefab(m_prefabMakerPage->getPrefabName());
+    }
+
+    bool RobotImporterWidget::CheckCyclicalDependency(AZ::IO::Path importedPrefabPath)
+    {
+        AzFramework::EntityContextId context_id;
+        EBUS_EVENT_RESULT(context_id, AzFramework::EntityIdContextQueryBus, GetOwningContextId);
+
+        AZ_Printf("CheckCyclicalDependency", "CheckCyclicalDependency %s", importedPrefabPath.Native().data());
+        auto focus_interface = AZ::Interface<AzToolsFramework::Prefab::PrefabFocusInterface>::Get();
+
+        if (!focus_interface)
+        {
+            ReportError(tr("Imported prefab could not be validated.\nImport aborted."));
+            return true;
+        }
+
+        auto focus_prefab_instance = focus_interface->GetFocusedPrefabInstance(context_id);
+
+        if (!focus_prefab_instance)
+        {
+            ReportError(tr("Imported prefab could not be validated.\nImport aborted."));
+            return true;
+        }
+
+        auto focus_prefab_filename = focus_prefab_instance.value().get().GetTemplateSourcePath();
```

Comment:
focus_prefab_filename should be lowerCamelCase per the style guide: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names
```suggestion
        auto focusPrefabFilename = focusPrefabInstance.value().get().GetTemplateSourcePath();
```